sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -506,33 +507,74 @@ public DirectoryAsyncClient getSubDirectoryClient(String subDirectoryName) {
-     * handles.
-     * @param recursive A boolean value that specifies if the operation should also apply to the files and
-     * subdirectories of the directory specified in the URI.
-     * @return The counts of number of handles closed
+     * @param handleId Handle ID to be closed.
+     * @return An empty response.
      */
-    public PagedFlux<Integer> forceCloseHandles(String handleId, boolean recursive) {
-        return forceCloseHandlesWithOptionalTimeout(handleId, recursive, null, Context.NONE);
+    public Mono<Void> forceCloseHandle(String handleId) {
+        return withContext(context -> forceCloseHandleWithResponse(handleId, context)).flatMap(FluxUtil::toMono);
     }
 
-    PagedFlux<Integer> forceCloseHandlesWithOptionalTimeout(String handleId, boolean recursive, Duration timeout,
-        Context context) {
+    /**
+     * Closes a handle on the directory. This is intended to be used alongside {@link #listHandles(Integer, boolean)}.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Force close handles returned by list handles.</p>
+     *
+     * {@codesnippet com.azure.storage.file.DirectoryAsyncClient.forceCloseHandleWithResponse#String}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/force-close-handles">Azure Docs</a>.</p>
+     *
+     * @param handleId Handle ID to be closed.
+     * @return A response that only contains headers and response status code.
+     */
+    public Mono<Response<Void>> forceCloseHandleWithResponse(String handleId) {
+        return withContext(context -> forceCloseHandleWithResponse(handleId, context));
+    }
+
+    Mono<Response<Void>> forceCloseHandleWithResponse(String handleId, Context context) {
+        return this.azureFileStorageClient.directorys().forceCloseHandlesWithRestResponseAsync(shareName, directoryPath,
+            handleId, null, null, snapshot, false, context)
+            .map(response -> new SimpleResponse<>(response, null));
+    }
+
+    /**
+     * Closes all handles opened on the directory at the service.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Force close all handles recursively.</p>
+     *
+     * {@codesnippet com.azure.storage.file.DirectoryAsyncClient.forceCloseAllHandles#boolean}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/force-close-handles">Azure Docs</a>.</p>
+     *
+     * @param recursive Flag indicating if the operation should apply to all subdirectories and files contained in the
+     * directory.
+     * @return The number of handles closed.
+     */
+    public Mono<Integer> forceCloseAllHandles(boolean recursive) {
+        return fluxContext(context -> forceCloseAllHandlesWithTimeout(recursive, null, context))
+            .reduce(0, Integer::sum);
```

Comment:
nit; alternatively we can use `count()` operator https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#count--

Reply:
Let's continue using reduce for now as we won't need to do another mapping to downcast.

Reply:
Sounds good.